Userland/Libraries/LibPDF/Document.cpp

```diff
@@ -34,20 +34,32 @@ String OutlineItem::to_string(int indent) const
     return builder.to_string();
 }
 
-Document::Document(const ReadonlyBytes& bytes)
-    : m_parser(Parser({}, bytes))
+RefPtr<Document> Document::create(const ReadonlyBytes& bytes)
 {
-    m_parser.set_document(this);
+    auto parser = adopt_ref(*new Parser({}, bytes));
+    auto document = adopt_ref(*new Document(parser));
 
-    VERIFY(m_parser.perform_validation());
-    auto [xref_table, trailer] = m_parser.parse_last_xref_table_and_trailer();
+    VERIFY(parser->perform_validation());
+    auto xref_table_and_trailer_opt = parser->parse_last_xref_table_and_trailer();
+    if (!xref_table_and_trailer_opt.has_value())
+        return {};
 
-    m_xref_table = xref_table;
-    m_trailer = trailer;
+    auto [xref_table, trailer] = xref_table_and_trailer_opt.value();
```

Comment:
```suggestion
    auto& [xref_table, trailer] = xref_table_and_trailer_opt.value();
```
or
```suggestion
    auto [xref_table, trailer] = xref_table_and_trailer_opt.release_value();
```

Reply:
This is also changed later, and the table becomes a `RefPtr`